src/actions/blocks.js

```diff
@@ -84,7 +84,7 @@ export const forgingTimesRetrieved = nextForgers => async (dispatch, getState) =
   latestBlocks.forEach((item, index) => {
     if (index > forgedInRoundNum && !forgingTimes[item.generatorPublicKey]) {
       // if I can't find it in the previous round
-      if (index >= forgedInRoundNum + 101 && latestBlocks.indexOf(item) === index) {
+      if (index >= forgedInRoundNum + 103 && latestBlocks.indexOf(item) === index) {
```

Comment:
I am looking at `latestBlocks.indexOf(item) === index` and I don't understand why is it there. This should always be true or am I missing something?

Reply:
Latest blocks are 206, 2 rounds. this checks if the item exists in the lest round or the one before. We can improve though.

Reply:
Okay but the `index` is already the _index_ at which the `item` is situated at in the `latestBlocks` array. i.e. `latestBlocks[index]` returns `item`. So, `latestBlocks.indexOf(item) === index` would always be true.

Reply:
Assume we have `arr = ['a', 'b', 'c, 'a']`, then `arr.forEach(item => arr.indexOf(item))` will return `[0, 1, 2, 0]`.